module_utils/network_lsr/argument_validator.py

```diff
@@ -549,58 +556,227 @@ def __init__(self):
             self,
             name="features",
             nested=[
-                ArgValidatorBool("esp-hw-offload", default_value=None),
-                ArgValidatorBool("esp-tx-csum-hw-offload", default_value=None),
-                ArgValidatorBool("fcoe-mtu", default_value=None),
+                ArgValidatorBool("esp_hw_offload", default_value=None),
+                ArgValidatorBool(
+                    "esp-hw-offload", default_value=None, deprecated_by="esp_hw_offload"
```

Comment:
@elvgarrui another thought: Maybe the necessary changes can be simplified by removing `default_value=None` from the items that are deprecated. Maybe it even makes sense to fail when `default_value` and `deprecated_by` are both specified.

Reply:
> @elvgarrui another thought: Maybe the necessary changes can be simplified by removing `default_value=None` from the items that are deprecated. Maybe it even makes sense to fail when `default_value` and `deprecated_by` are both specified.

Actually it seems everything needs to have a default value so it might be more complicated to change this.

Reply:
Another thought: We can probably introduce a new class `ArgValidatorDeprecated` and use that here instead of `ArgValidatorBool` for the deprecated values since they are not used for anything and do not need a default value.